### PR TITLE
Change RBAC storage version to v1 for 1.9

### PR DESCRIPTION
**Pull request by liggitt**
v1 was introduced in 1.8, but storage version remained at v1beta1 to accommodate HA rolling upgrades. in 1.9, we can change the persisted and preferred version to v1

```release-note
RBAC objects are now stored in etcd in v1 format. After completing an upgrade to 1.9, RBAC objects (Roles, RoleBindings, ClusterRoles, ClusterRoleBindings) should be migrated to ensure all persisted objects are written in `v1` format, prior to `v1alpha1` support being removed in a future release.
```

### CODE REVIEW

**Comment by liggitt** [comment]
cc @kubernetes/sig-auth-api-reviews @kubernetes/sig-auth-pr-reviews

**Comment by enj** [comment]
/lgtm

No types are being removed so this complies with #52185

@lavalamp @thockin

**Comment by liggitt** [comment]
```
I0923 22:40:39.353] Ran 315 of 693 Specs in 2172.830 seconds
I0923 22:40:39.353] FAIL! -- 315 Passed | 0 Failed | 0 Pending | 378 Skipped 
I0923 22:40:39.353] 
I0923 22:40:39.354] Ginkgo ran 1 suite in 36m15.611467073s
I0923 22:40:39.354] Test Suite Failed
```
...
/retest

**Comment by liggitt** [comment]
Kernel panic flake
/retest

File: federation/pkg/kubefed/join_test.go [hunk 1]
```diff
@@ -509,10 +509,10 @@ func fakeJoinTargetClusterFactory(clusterName, clusterCtx, dnsProvider, tmpDirPa
 	apiGroupList.Groups = append(apiGroupList.Groups, testGroup)
 	if isRBACAPIAvailable {
 		rbacGroup := metav1.APIGroup{
-			Name: rbacv1beta1.GroupName,
+			Name: rbacv1.GroupName,
 			Versions: []metav1.GroupVersionForDiscovery{
 				{
-					GroupVersion: rbacv1beta1.GroupName + "/v1beta1",
+					GroupVersion: rbacv1.GroupName + "/v1beta1",
```
Comment (sttts) [inline federation/pkg/kubefed/join_test.go]: /v1 ?

**Comment by sttts** [comment]
One little test typo. Otherwise, lgtm.

**Comment by liggitt** [comment]
comment addressed

**Comment by liggitt** [comment]
/assign @mikedanese 
for cluster/update-storage-objects.sh change

/assign @nikhiljindal
for federation test update approval

**Comment by liggitt** [comment]
/unassign @nikhiljindal 
/assign @quinton-hoole 
for federation unit test change approval

**Pull request merged**